asf_hp_cost_estimator_model/pipeline/hyperparameter_tuning/tune_hyperparameters.py

```diff
@@ -0,0 +1,186 @@
+)
+from asf_hp_cost_estimator_model.getters.data_getters import (
+    get_enhanced_installations_data,
+)
+from asf_hp_cost_estimator_model.pipeline.data_processing.process_location_data import (
+    get_postcodes_data,
+)
+from asf_hp_cost_estimator_model import config
+from asf_hp_cost_estimator_model.pipeline.data_processing.process_cpi import (
+    get_df_quarterly_cpi_with_adjustment_factors,
+)
+from asf_hp_cost_estimator_model.getters.data_getters import get_cpi_data
+from asf_hp_cost_estimator_model.utils.model_evaluation_utils import compute_metrics
+
+
+def argparse_setup():
+    """
+    Sets up the command line argument parser to allow for quantile specification
+    used for estimating a cost prediction interval for an air source heat pump.
+    """
+    import argparse
+
+    parser = argparse.ArgumentParser(
+        description="Fit a model to estimate the cost of an air source heat pump."
+    )
+    parser.add_argument(
+        "--lower_quantile",
+        type=float,
+        default=0.1,
+        help="Lower quantile for cost estimation.",
+    )
+    parser.add_argument(
+        "--upper_quantile",
+        type=float,
+        default=0.9,
+        help="Upper quantile for cost estimation.",
+    )
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    args = argparse_setup()
+    lower_quantile = args.lower_quantile
+    upper_quantile = args.upper_quantile
+
+    # Load and process data
+    mcs_epc_data = get_enhanced_installations_data()
+    cpi_05_3_df = get_cpi_data()
+    cpi_quarterly_df = get_df_quarterly_cpi_with_adjustment_factors(
+        ref_year=config["cpi_data"]["cpi_reference_year"],
+        cpi_df=cpi_05_3_df,
+        cpi_col_header=config["cpi_data"]["cpi_column_header"],
+    )
+    postcodes_data = get_postcodes_data()
+
+    model_data = process_data_before_modelling(
+        mcs_epc_data=mcs_epc_data,
+        postcodes_data=postcodes_data,
+        cpi_quarterly_df=cpi_quarterly_df,
+        exclusion_criteria_dict=config["exclusion_criteria"],
+        winsorise=config["winsorise_outliers"],
+        min_date=config["min_date"],
+    )
+
+    # Define features and target
+    numeric_features = config["numeric_features"]
+    original_categorical_features = config["categorical_features_to_dummify"]
+    categorical_features = config["categorical_features"]
+    target_feature = config["target_feature"]
+
+    # Preparing data for modelling
+    X = model_data[numeric_features + categorical_features]
+    y = model_data[target_feature].values.ravel()
+
+    # Split the data into training and test sets
+    x_train, x_test, y_train, y_test = train_test_split(
+        X, y, test_size=0.2, random_state=config["random_state"]
+    )
+
+    # Define the parameter grid for hyperparameter tuning
+    param_grid = dict(
+        learning_rate=[0.01, 0.05, 0.1, 0.2],
+        max_depth=[3, 5, 10, 20],
+        min_samples_leaf=[1, 5, 10, 100, 10000],
```

Comment:
```suggestion
        min_samples_leaf=[1, 5, 10, 100, 1000],
```
Is this meant to be 1000?